src/RuntimePluginLoader.cpp

```diff
@@ -0,0 +1,109 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "RAJA/util/RuntimePluginLoader.hpp"
+
+#include <dlfcn.h>
+#include <dirent.h>
+
+inline
+bool
+isSharedObject(const std::string filename)
+{
+  return (filename.size() > 3 && !filename.compare(filename.size() - 3, 3, ".so"));
```

Comment:
What level of cross platform (windows) support are we targeting with this capability?

Reply:
Some of the current code is more geared towards linux at the moment, but cross platform support can pretty easily be added. I'm not too experienced with windows shared object libraries and their differences from linux, so that function is somewhat of a placeholder for something a little more versatile in terms of cross platform support. I do apologize for the lack of commenting about that.

Reply:
While on the topic of cross platform support for windows, @willkill07 mentioned this repository:
https://github.com/dlfcn-win32/dlfcn-win32?fbclid=IwAR3OMQuPd3OW-jPQ7SboTZikd3yeDhnoMwlIt9mRb0-yoFacWE4-DTzlC3w

Which provides wrapper functions for dlopen and dlsym compatible with windows when included in the CMake. I haven't gotten around the implementing it yet, but it's worth noting!